rpc_new/provider/smoldot.ts

```diff
@@ -0,0 +1,92 @@
+import { start } from "../../deps/smoldot.ts";
+import { Chain, Client, ClientOptions } from "../../deps/smoldot/client.d.ts";
+import * as msg from "../messages.ts";
+import { nextIdFactory, Provider, ProviderConnection, ProviderListener } from "./base.ts";
+import { ProviderCloseError, ProviderHandlerError, ProviderSendError } from "./errors.ts";
+
+// TODO
+type SmoldotSendErrorData = any;
+type SmoldotHandlerErrorData = any;
+type SmoldotCloseErrorData = any;
+
+let client: undefined | Client;
+const connections = new Map<string, SmoldotProviderConnection>();
+class SmoldotProviderConnection
+  extends ProviderConnection<Chain, SmoldotSendErrorData, SmoldotHandlerErrorData>
+{}
+
+const nextId = nextIdFactory();
+
+export const smoldotProvider: Provider<
+  string,
+  SmoldotSendErrorData,
+  SmoldotHandlerErrorData,
+  SmoldotCloseErrorData
+> = (chainSpec, listener) => {
+  return {
+    nextId,
+    send: (message) => {
+      (async () => {
+        const conn = await connection(chainSpec, listener);
+        try {
+          return conn.inner.sendJsonRpc(JSON.stringify(message));
+        } catch (e) {
+          return new ProviderSendError(e);
+        }
+      })();
```

Comment:
These errors are getting sent to the void

Reply:
Whoops. Thought I had a TODO note there.